Infusionsoft/InvoiceService.php

```diff
@@ -37,11 +37,16 @@ public static function chargeInvoiceArbitraryAmount($contactId, $invoiceId, $car
     public static function chargeInvoiceArbitraryAmountUsingPayPlan($invoiceId, $cardId, $amount, $merchantAccountId, $paymentNotes = 'API Arbitrary Payment', $numberOfPaymentsForChargePaymentPlan = 1, $daysBetweenPaymentsForChargePaymentPlan = 1){
 
         $result = false;
+        $today = new DateTime('now', new DateTimeZone('America/New_York'));
+        $today = $today->format('Y-m-d');
 
         //Get Invoice info so we can set the temporary PayPlan to match the amount we want to charge
+
+        //In rare occassions, PHP can interpret the $amount or the $invoice->TotalPaid as a float, messing up the comparisons
+        // For this reason, we force $amount and TotalPaid to float, then round the sum to 2 places and use that value for the comparisons.
         $invoice = new Infusionsoft_Invoice($invoiceId);
-        if ($amount + $invoice->TotalPaid <= $invoice->InvoiceTotal){
-            $temporaryFirstPayment = $amount + $invoice->TotalPaid;
+        if (round((float)$amount + (float)$invoice->TotalPaid, 2) <= $invoice->InvoiceTotal){

```

Comment:
I'm a little concerned about the rounding.  The comment doesn't make much sense as "In rare occassions, PHP can interpret the $amount or the $invoice->TotalPaid as a float, messing up the comparisons" so you don't want it to be interpreted as a float but you are casting to a float? 

I ran into this problem at Infusionsoft where the InvoiceTotal would be a 3 place decimal because of a percentage discount so it would never be marked as "PAID" since there was a fraction of a cent left over.  I kind of think this fix is a sledge hammer to smash a cockroach.  Can you provide some more details?